components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1437,8 +1417,59 @@ void ConversationHandler::OnEngineCompletionComplete(
         SetAPIError(mojom::APIError::None);
       }
     }
+    CompleteGeneration(false);
+    return;
   }
 
+  // Handle success, which might mean do nothing much since all data was passed
+  // in the streaming "received" callback.
+  DVLOG(2) << __func__ << ": With value";
+  if (result->event && result->event->is_completion_event() &&
+      !result->event->get_completion_event()->completion.empty()) {
+    UpdateOrCreateLastAssistantEntry(std::move(*result));
+  } else {
+    // This is a workaround for any occasions where the engine returns
+    // a success but there was no new entry.
+    if (needs_new_entry_) {
+      SetAPIError(mojom::APIError::ConnectionIssue);
+      CompleteGeneration(false);
```

Comment:
Changed this to be consistent with other error cases, no MaybePopPendingRequests here, it would be called when RetryAPI is called via UI with a successful response.